### PR TITLE
feat(3498) Remove \n from content log end

### PR DESCRIPTION
**Pull request by Y-Henry**

### CODE REVIEW

File: pkg/entry/mongo/document.go [hunk 1]
```diff
@@ -121,13 +122,19 @@ func (d *JobLogDocument) Populate(ctx context.Context, ts time.Time, record map[
 	return d.generateObjectID()
 }
 
+func cleanLogContent(content string) string {
+	endOfLineRegexp := regexp.MustCompile(`(\n)+$`)
+	return endOfLineRegexp.ReplaceAllString(content, "")
```
Comment (holyhope) [inline pkg/entry/mongo/document.go]: Tip, if you use regex, please compile them on start, not in each `cleanLogContent` calls.

Anyway, I suggest you to use: [`strings.TrimSuffix`](https://pkg.go.dev/strings#TrimSuffix)

```suggestion
	return strings.TrimSuffix(content, "\n")
```

PS Here only a single `\n` is stripped, IMO this is what we want.

File: pkg/entry/mongo/document.go [hunk 2]
```diff
@@ -121,13 +122,19 @@ func (d *JobLogDocument) Populate(ctx context.Context, ts time.Time, record map[
...
 	if err != nil {
 		return fmt.Errorf("get logger: %w", err)
 	}
 
-	d.Log, err = parse.ExtractStringValue(record, LogKey)
+	logContent, err := parse.ExtractStringValue(record, LogKey)
+	d.Log = cleanLogContent(logContent)
 	if err != nil {
 		if !errors.Is(err, &parse.ErrKeyNotFound{
 			LookingFor: LogKey,
```
Comment (holyhope) [inline pkg/entry/mongo/document.go]: Never ever use a value before checking the associated error!

File: pkg/entry/mongo/document_test.go [hunk 3]
```diff
@@ -126,4 +126,54 @@ var _ = Describe("Convert document", func() {
...
+		It("Should remove multiple \\n from log end", func() {
+			entry[mongo.LogKey] = stringEntry("log\n\n\n")
+
+			d, err := mongo.Convert(ctx, time.Now(), entry)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(d).ToNot(BeNil())
+			Expect(d).To(BeAssignableToTypeOf(&mongo.AppLogDocument{}))
+			document := d.(*mongo.AppLogDocument)
+			Expect(document.Log).To(BeEquivalentTo(stringEntry("log")))
+		})
```
Comment (holyhope) [inline pkg/entry/mongo/document_test.go]: You sure that is the expected behavior?

**Review by holyhope — CHANGES_REQUESTED** [review]

File: pkg/entry/mongo/document_test.go (hunk 3 above)
Comment (Y-Henry) [reply pkg/entry/mongo/document_test.go]: Not really, It was an unexpected behavior. Using a regexp, it was easy to protect against this unexpected event. I remove it

File: pkg/entry/mongo/document.go (hunk 2 above)
Comment (Y-Henry) [reply pkg/entry/mongo/document.go]: No excuse to put this line here, I agree :D

File: pkg/entry/mongo/document.go (hunk 1 above)
Comment (Y-Henry) [reply pkg/entry/mongo/document.go]: Fixed

**Review by holyhope — APPROVED** [review]

**Review by fabienmifsud — APPROVED** [review]

**Pull request merged**